p2p/conn/connection.go

```diff
@@ -220,16 +225,23 @@ func (c *MConnection) stopServices() (alreadyStopped bool) {
 
 	select {
 	case <-c.quitSendRoutine:
-		// already quit via FlushStop or OnStop
+		// already quit
 		return true
 	default:
 	}
 
+	// inform the recvRouting that we are shutting down
+	select {
```

Comment:
does this need to be sent on here or can we just close it ?

Reply:
This needs to be sent here to inform the receive routine that it needs to stop. The is no mechanism in go (that I have found) to receive socket events therefore you need to roll your own notifications to the go routine for receive that was setup previously.

Reply:
Oh, you can use `close(ch)` as a way to send on a channel one time, and to have many receivers be able to receive the signal - https://golang.org/ref/spec#Close

Reply:
But does the signal need to be sent here, or can it be sent after the block of `.Stop()` calls? Calling `close(ch)` will send a signal on the channel.

Reply:
Ah - I did not know that about channels. I have removed this select that manually sent a bool in favor of closing the channel (already done lower in the same method). I have changed the channel to a `struct{}` type since nothing is sent on it anymore.